compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -486,7 +488,7 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
                 }
             }
             let llval = const_llval.unwrap_or_else(|| {
-                let load = self.load(place.llval, place.align);
+                let load = self.load(self.backend_type(place.layout), place.llval, place.align);
```

Comment:
Nit: Since this is within LLVM specific code, I would probably call the more direct `place.layout.llvm_type(self)` here, as is done throughout the rest of this crate.